src/v/cloud_storage/tests/topic_manifest_test.cc

```diff
@@ -118,6 +166,82 @@ SEASTAR_THREAD_TEST_CASE(update_serialize_update_same_object) {
     BOOST_REQUIRE(m == restored);
 }
 
+SEASTAR_THREAD_TEST_CASE(min_config_update_all_fields_correct) {
+    topic_manifest m;
+    m.update(make_manifest_stream(min_topic_manifest_json)).get();
+
+    auto topic_config = m.get_topic_config().value();
+    BOOST_REQUIRE_EQUAL(m.get_revision(), model::initial_revision_id(0));
+    BOOST_REQUIRE_EQUAL(
+      topic_config.tp_ns,
+      model::topic_namespace(
+        model::ns("test-namespace"), model::topic("test-topic")));
+    BOOST_REQUIRE_EQUAL(topic_config.partition_count, 32);
+    BOOST_REQUIRE_EQUAL(topic_config.replication_factor, 3);
+    BOOST_REQUIRE(!topic_config.properties.compression);
+    BOOST_REQUIRE(!topic_config.properties.cleanup_policy_bitflags);
+    BOOST_REQUIRE(!topic_config.properties.compaction_strategy);
+    BOOST_REQUIRE(!topic_config.properties.timestamp_type);
+    BOOST_REQUIRE(!topic_config.properties.segment_size);
+    BOOST_REQUIRE(
+      !topic_config.properties.retention_bytes.is_disabled()
+      && !topic_config.properties.retention_bytes.has_value());
+    BOOST_REQUIRE(
+      !topic_config.properties.retention_duration.is_disabled()
+      && !topic_config.properties.retention_duration.has_value());
+}
+
+SEASTAR_THREAD_TEST_CASE(full_config_update_all_fields_correct) {
+    topic_manifest m;
+    m.update(make_manifest_stream(full_topic_manifest_json)).get();
+
+    auto topic_config = m.get_topic_config().value();
+    BOOST_REQUIRE_EQUAL(m.get_revision(), model::initial_revision_id(1));
+    BOOST_REQUIRE_EQUAL(
+      topic_config.tp_ns,
+      model::topic_namespace(
+        model::ns("full-test-namespace"), model::topic("full-test-topic")));
+    BOOST_REQUIRE_EQUAL(topic_config.partition_count, 64);
+    BOOST_REQUIRE_EQUAL(topic_config.replication_factor, 6);
+    BOOST_REQUIRE_EQUAL(
+      topic_config.properties.compression.value(), model::compression::snappy);
+    BOOST_REQUIRE_EQUAL(
+      topic_config.properties.cleanup_policy_bitflags.value(),
+      model::cleanup_policy_bitflags::deletion
+        | model::cleanup_policy_bitflags::compaction);
+    BOOST_REQUIRE_EQUAL(
+      topic_config.properties.compaction_strategy.value(),
+      model::compaction_strategy::offset);
+    BOOST_REQUIRE_EQUAL(
+      topic_config.properties.timestamp_type.value(),
+      model::timestamp_type::append_time);
+    BOOST_REQUIRE_EQUAL(topic_config.properties.segment_size.value(), 1234);
+    BOOST_REQUIRE_EQUAL(topic_config.properties.retention_bytes.value(), 42342);
+    BOOST_REQUIRE_EQUAL(
+      topic_config.properties.retention_duration.value(),
+      std::chrono::milliseconds(36000000000));
```

Comment:
I think you can increase test coverage to include the new validation checks added which throw. Boost has a cool utility for this:

```c++
BOOST_REQUIRE_EXCEPTION(
   some_expression_that_throws(),
   std::runtime_error,
   [](std::runtime_error ex) {
      return ex.what() == ""Wrong cleanup_policy_bitflags value: XXX";
   });
```

where you can test that JSON parsing throws one of your new errors on bad input, and then get even more precise and show that it contains the expected error.

Reply:
Looks good, used it!